File: src/main/js/ch1/look-and-say.js
function ant(n) {
  let s = "1"
  for (let i = 0; i < n; i++) {
    s = next(s)
  }
  return s
}

function next(ns) {
  let result = ""
  for (let i = 0; i < ns.length; i++) {
    let length = 1
    while (i+1 < ns.length && ns[i] === ns[i+1]) {
      length++
      i++
    }
    result += length + ns[i]
  }
  return result
}

console.log(ant(10))

File: src/main/java/ch1/LookAndSay.java
package ch1;

public class LookAndSay {
  public static void main(String... args) {
    String s = "1";
    for (int line = 0; line < 20; line++) {
      System.out.println(s);
      int length = 1;
      char head = s.charAt(0);
      String result = "";
      for (int i = 1; i < s.length(); i++) {
        if (s.charAt(i) == head) {
          length++;
        } else {
          result += length;
          result += head;
          length = 1;
          head = s.charAt(i);
        }
      }
      result += length;
      result += head;
      s = result;
    }
    System.out.println(s);
  }
}